### PR TITLE
Translate BSL to Repoistory Parameters.

### PR DESCRIPTION
**Pull request by deepakkinni**
1. Retrieve the BSL for the specific backup request.
2. From the BSL extract all the necessary aws params- region, bucket, s3ForcePathStyle, s3Url
3. Among them, extract the current profile in use, this is specified in bsl.Spec.Config["profile"]
4. The above profile determines which credentials will be used from the cloud-credential file.
5. Retrieve the "cloud-credential" Secret from velero namespace.
6. secret.Data holds the whole credential file in byte format.
7. Create a temp file to move the credentials into it, this is done so that we can use pre-existing aws credential
apis to extract the access key/secret.
8. Use credentials.NewSharedCredentials(tmpfileName, bsl-profile) to extract the relevant key/secret.
9. Store them in repository parameters.
10. Use the repo parameters everywhere the s3 repo is being initialized, currently this happens in GetS3PETMFromParamsMap.

Testing.
1. Read bsl name and extracted the credentials for the bsl in the guest cluster.
2. Triggered BRC creation in the guest.
3. Verified that the BR in supervisor cluster has the credentials populated.

Signed-off-by: Deepak Kinni <dkinni@vmware.com>

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/62/

File: pkg/plugin/backup_pvc_action_plugin.go [hunk 1]
```diff
@@ -70,8 +70,17 @@ func (p *NewPVCBackupItemAction) Execute(item runtime.Unstructured, backup *vele
...
 	}
 
+
+
 	p.Log.Info("Claiming backup repository for snapshot")
-	repositoryParameters := make(map[string]string) // TODO: retrieve the real object store config from velero storage location
+	bslName := backup.Spec.StorageLocation
+	repositoryParameters := make(map[string]string)
+
+	err = utils.RetrieveParamsFromBSL(repositoryParameters, bslName, restConfig, p.Log)
```
Comment (swgupta-vmw) [inline pkg/plugin/backup_pvc_action_plugin.go:79]: Should we add a check for local mode here? repositoryParameters can remain empty in local mode.

File: pkg/utils/constants.go [hunk 2]
```diff
@@ -83,7 +87,8 @@ const (
 const (
 	// DefaultNamespace is the Kubernetes namespace that is used by default for
 	// the Velero server and API objects.
-	DefaultNamespace = "velero"
+	DefaultNamespace  = "velero"
+	DefaultSecretName = "cloud-credentials"
```
Comment (swgupta-vmw) [inline pkg/utils/constants.go:91]: Can we give a more specific name, eg CloudCredentialSecretName. There are several secrets in the velero namespace, so not sure if we should call this default secret.

**Comment by svcbot-qecnsdp** [comment]
Congratulations !! 

All the pre-check tests passed
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/69/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/70/

**Pull request merged**